Detectors/TPC/workflow/src/TPCIntegrateDCASpec.cxx

```diff
@@ -0,0 +1,265 @@
+  float mHBScaling = 1;                                   ///< fraction of a TF filled with data (in case only fraction of TF stored)
+  const bool mDisableWriter{false};                       ///< flag if no ROOT output will be written
+  ITPCDCA mBufferDCA;                                     ///< buffer for integrate DCAs
+  std::vector<RobustAverage> mAvgA;                       ///< for averaging the DCAs for A-side
+  std::vector<RobustAverage> mAvgC;                       ///< for averaging the DCAs for C-side
+  std::shared_ptr<o2::base::GRPGeomRequest> mCCDBRequest; ///< info for CCDB request
+  int mContinuousMaxTimeBin{-1};                          ///< max time bin of clusters
+  bool mInitICCBuffer{true};                              ///< flag for initializing ICCs only once
+  int mNTSPerSlice{1};                                    ///< number of time stamps per slice
+  int mNMaxTracks{-1};                                    ///< maximum number of tracks to process
+  float mMinMom{1};                                       ///< minimum accepted momentum
+  int mMinNCl{80};                                        ///< minimum accepted number of tracks
+  float mMaxEta{1};                                       ///< maximum eta
+  float mCoarseStep{1};                                   ///< coarse step during track propagation
+  float mFineStep{0.005};                                 ///< fine step during track propagation
+  float mCutDCA{5};                                       ///< cut on the abs(DCA-median)
+
+  bool acceptTrack(const o2::tpc::TrackTPC& track)
+  {
+    if ((track.getP() < mMinMom) || (track.getNClusters() < mMinNCl) || track.hasBothSidesClusters() || std::abs(track.getTgl()) > mMaxEta) {
+      return false;
+    }
+    return true;
+  }
+
+  void fillDCA(o2::tpc::TrackTPC track)
+  {
+    // propagate track to the DCA and fill in slice
+    auto propagator = o2::base::Propagator::Instance();
+
+    // propagate track to DCA
+    o2::gpu::gpustd::array<float, 2> dca;
+    const o2::math_utils::Point3D<float> refPoint{0, 0, 0};
+    if (!propagator->propagateToDCABxByBz(refPoint, track, mCoarseStep, o2::base::Propagator::MatCorrType::USEMatCorrLUT, &dca)) {
+      return;
+    }
+
+    if (!propagator->propagateToDCABxByBz(refPoint, track, mFineStep, o2::base::Propagator::MatCorrType::USEMatCorrLUT, &dca)) {
+      return;
+    }
+
+    const unsigned int sliceInTF = track.getTime0() / mNTSPerSlice;
+    if (sliceInTF < mNSlicesTF) {
+      // DCAr
+      const float dcar = dca[0];
+
+      // weight of DCAr
+      const float resCl = track.getNClusters() / Mapper::PADROWS;
+      const float resCl2 = resCl * resCl;
+      const float fB = 1 / (resCl2 * track.getPt());
+
+      const float fA = 0.15;
+      const float dcarW = fA * fA + 0.2 * fB * fB; // Rms2 ~ 0.15^2 +  k/(L^2*pt) →    0.15**2 +  (0.2/((NCl/152)^2*pt)^2);
```

Comment:
Hi @miranov25 ,
can you check if the formula for the weight is correct

Reply:
Is there rescl2 squared? Should be only squared not Power 4

Reply:
After discussion with @matthias-kleiner 
```
      const float resCl = track.getNClusters() / Mapper::PADROWS;
      const float fB = 0.2 / (resCl * track.getPt());

      const float fA = 0.15;
      const float dcarW = std::sqrt(fA * fA + fB * fB); // Rms2 ~ 0.15^2 +  k/(L^2*pt) →    0.15**2 +  (0.2/((NCl/152)^2*pt)^2);
```